src/identity.js

```
@@ -1663,106 +1622,63 @@ export default function Identity(mpInstance) {
-                        ? previousUIByNameCopy
-                        : incomingMpidUIByNameCopy
+                    uiByName
                 );
             }
 
             if (callback) {
-                if (xhr.status === 0) {
-                    mpInstance._Helpers.invokeCallback(
-                        callback,
-                        HTTPCodes.noHttpCoverage,
-                        identityApiResult || null,
-                        newUser
-                    );
-                } else {
-                    mpInstance._Helpers.invokeCallback(
-                        callback,
-                        xhr.status,
-                        identityApiResult || null,
-                        newUser
-                    );
-                }
-            } else {
-                if (
-                    identityApiResult &&
-                    identityApiResult.errors &&
-                    identityApiResult.errors.length
-                ) {
-                    mpInstance.Logger.error(
-                        'Received HTTP response code of ' +
-                            xhr.status +
-                            ' - ' +
-                            identityApiResult.errors[0].message
-                    );
-                }
+                const callbackCode =
+                    xhr.status === 0 ? HTTPCodes.noHttpCoverage : xhr.status;
+
+                mpInstance._Helpers.invokeCallback(
+                    callback,
+                    callbackCode,
+                    identityApiResult || null,
+                    newUser
+                );
+            } else if (
+                identityApiResult &&
+                !isEmpty(identityApiResult.errors)
+            ) {
+                mpInstance.Logger.error(
+                    'Received HTTP response code of ' +
+                        xhr.status +
+                        ' - ' +
+                        identityApiResult.errors[0].message
```

Can we add a go link for figuring out why we'd only want to return the 1st error here?  It could start with figuring out what potential errors would be returned.